framework/include/kernels/ScalarKernelBase.h

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ResidualObject.h"
+#include "ScalarCoupleable.h"
+#include "MooseVariableScalar.h"
+
+// Forward declarations
+class ScalarKernelBase;
+
+template <>
+InputParameters validParams<ScalarKernelBase>();
```

Comment:
can remove

Reply:
Lesson learned. I should not have done `ScalarKernel` -> `ScalarKernelBase` and then made `ScalarKernel`; rather I should have just kept `ScalarKernel` and made `ScalarKernelNotBase` - that way I don't have to clean up the original class 😛

Reply:
Well you can note that I didn't ask you to remove this from `ScalarKernel`...

Reply:
By "clean up the original class" I meant "clean up code moved from the original class into the new base class". But nevermind, not a gripe, just another failed joke - I'm at like 82% joke success.